app/gui/src/dashboard/components/dashboard/column/PathColumn.tsx

```diff
@@ -65,11 +71,16 @@ export default function PathColumn(props: AssetColumnProps) {
     const targetDirectoryNode = getAssetNodeById(targetDirectory)
 
     if (targetDirectoryNode == null && rootDirectoryInThePath.categoryId != null) {
-      // We reassign the variable only to make TypeScript happy here.
-      const categoryId = rootDirectoryInThePath.categoryId
-
-      setCategory(categoryId)
-      setExpandedDirectoryIds(pathToDirectory.map(({ id }) => id).concat(targetDirectory))
+      setCategory(rootDirectoryInThePath.categoryId)
+      const expandedDirectories = structuredClone(driveStore.getState().expandedDirectories)
```

Comment:
This is not good, this is a common thing to change and instead of encapsulating it behind a function/hook, we expose the internals outside.

Ideally, changing the expanded directories _should not_ affect other components, because the logic should be encapsilated. Otherwise we violate SOLID principles

Reply:
> should not affect other components

wdym?

Reply:
We're changing the _internal_ structure of the data. If everything is encapsulated correctly, none of the consumers will be affected by this change. Internal change doesn't affect the API